src/index.ts

```diff
@@ -1,33 +1,31 @@
-import { Users } from './services/users';
-import { Logger } from './services/logger';
+import { createIoCContainer } from "./ioc";
 
-import type { User, ApiConfig } from './types';
+import type { User } from "./types";
 
-const renderUsers = async (config: ApiConfig) => {
-  const usersService = new Users(config);
+const ioc = createIoCContainer();
+
+const renderUsers = async () => {
+  const usersService = ioc.resolve("users");
   const users = await usersService.getUsers();
 
-  const listNode = document.getElementById('users-list');
+  const listNode = document.getElementById("users-list");
 
   users.forEach((user: User) => {
-    const listItemNode = document.createElement('li');
+    const listItemNode = document.createElement("li");
 
     listItemNode.innerHTML = user.name;
     listNode.appendChild(listItemNode);
   });
 };
 
 const app = () => {
-  const config = (window as any).__CONFIG__;
-  delete (window as any).__CONFIG__;
```

Comment:
you shouldn’t move the config code to the ioc container, since you can register config here:

`ioc.register("apiConfig", config.api);`